src/xdotoolify.js

```diff
@@ -87,21 +91,48 @@ var _getElementAndBrowserRect = async function(page, selector) {
 };
 
 var _getElementAndBrowserScreenRect = async function(page, selector) {
-  // TODO: only works in firefox
+  let ret = null;
+
   try {
-    var ret = await _getElementAndBrowserRect(page, selector);
+    ret = await _getElementAndBrowserRect(page, selector);
   } catch (e) {
     console.warn('getElementScreenRect failed for');
     console.warn(selector);
     throw e;
   }
+
   return page.executeScript(function(_ret) {
     _ret.rect.x += window.mozInnerScreenX;
     _ret.rect.y += window.mozInnerScreenY;
     return _ret;
   }, ret);
 };
 
+var waitUntilElementIsAvailable = async function(page, selector, timeout) {
+  var startTime = Date.now();
+  var curTime = Date.now();
+
+  while (curTime - startTime <= timeout) {
+    var element = await page.executeScript(function(_selector) {
+      var elem = null;
+      if (Array.isArray(_selector)) {
+        elem = document.querySelectorAll(_selector[0]);
+      }
+      else {
+        elem = document.querySelector(_selector);
+      }
+      return elem;
+    }, selector);
+    if (element != null) break;
```

Comment:
please use full braces even for one-line ifs

also use triple equal vs double equal so you get strict comparison
```
if (element !== null) {
  break;
}
```